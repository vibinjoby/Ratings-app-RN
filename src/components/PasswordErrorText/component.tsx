import React from 'react'
import { View, Text, Image } from 'react-native'

import styles from './styles'
export interface PasswordErrorTextProps {
  inputValue: any
}

const PasswordErrorText: React.FC<PasswordErrorTextProps> = ({
  inputValue,
}: PasswordErrorTextProps) => {
  const successImg = require('../../assets/success-img.png')
  const errorImg = require('../../assets/error.png')

  const isUpperCase = (string: string) => /(.*[A-Z].*)/.test(string)
  const isLowerCase = (string: string) => /(.*[a-z].*)/.test(string)
  return (
    <View>
      <Text style={styles.headerTxt}>Password must contain:</Text>

      <View style={styles.container}>
        <Image
          testID="lessCharacters"
          source={inputValue.length >= 8 ? successImg : errorImg}
          style={styles.img}
        />
        <Text style={styles.subHeader}>At least 8 characters</Text>
      </View>
      <View style={styles.container}>
        <Image
          testID="minUppercase"
          source={inputValue.length > 0 && isUpperCase(inputValue) ? successImg : errorImg}
          style={styles.img}
        />
        <Text style={styles.subHeader}>At least 1 uppercase letter</Text>
      </View>
      <View style={styles.container}>
        <Image
          testID="minLowercase"
          source={inputValue.length > 0 && isLowerCase(inputValue) ? successImg : errorImg}
          style={styles.img}
        />
        <Text style={styles.subHeader}>At least 1 lowercase letter</Text>
      </View>
      <View style={styles.container}>
        <Image
          testID="minSplCharacter"
          source={
            inputValue.match(/^.*[!@#$%^&*()_+\-=\[\]{};':"\\|,.<>\/?].*$/) ? successImg : errorImg
          }
          style={styles.img}
        />
        <Text style={styles.subHeader}>At least 1 special character(#,%,&)</Text>
      </View>
      <View style={styles.container}>
        <Image
          source={inputValue.match(/\d/) ? successImg : errorImg}
          style={styles.img}
          testID="minNumber"
        />
        <Text style={styles.subHeader}>At least 1 number (0-9)</Text>
      </View>
    </View>
  )
}

export default PasswordErrorText
