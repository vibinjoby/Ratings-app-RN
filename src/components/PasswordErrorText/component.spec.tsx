import 'react-native'
import React from 'react'
import { render } from '@testing-library/react-native'

import PasswordErrorText from './component'

describe('PasswordErrorText', () => {
  it('should render', () => {
    const { toJSON } = render(<PasswordErrorText inputValue="" />)
    expect(toJSON()).toMatchSnapshot()
  })
})
