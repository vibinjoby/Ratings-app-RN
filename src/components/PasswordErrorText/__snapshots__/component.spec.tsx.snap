// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordErrorText should render 1`] = `
<View>
  <Text
    style={
      Object {
        "color": "rgba(30, 33, 76, 0.6)",
      }
    }
  >
    Password must contain:
  </Text>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/error.png",
        }
      }
      style={
        Object {
          "height": 10,
          "marginRight": 3,
          "marginTop": 3,
          "width": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgba(30, 33, 76, 0.4)",
        }
      }
    >
      At least 8 characters
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/error.png",
        }
      }
      style={
        Object {
          "height": 10,
          "marginRight": 3,
          "marginTop": 3,
          "width": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgba(30, 33, 76, 0.4)",
        }
      }
    >
      At least 1 uppercase letter
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/error.png",
        }
      }
      style={
        Object {
          "height": 10,
          "marginRight": 3,
          "marginTop": 3,
          "width": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgba(30, 33, 76, 0.4)",
        }
      }
    >
      At least 1 lowercase letter
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/error.png",
        }
      }
      style={
        Object {
          "height": 10,
          "marginRight": 3,
          "marginTop": 3,
          "width": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgba(30, 33, 76, 0.4)",
        }
      }
    >
      At least 1 special character(#,%,&)
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/error.png",
        }
      }
      style={
        Object {
          "height": 10,
          "marginRight": 3,
          "marginTop": 3,
          "width": 10,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "rgba(30, 33, 76, 0.4)",
        }
      }
    >
      At least 1 number (0-9)
    </Text>
  </View>
</View>
`;
